### ISSUE TITLE
YOLO8 OBB Prediction not working correctly!

### ISSUE THREAD

**Issue by nhatdtn** [ultralytics/ultralytics#14625]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I train yolov8 obb nano on my custom dataset, the visualized output of detected box is good on both validation set and test set when I use model.val(). However, when I use prediction code, the result is very poor on both validation set and test set, the detected box is completely wrong and different with detected box when using Val function. Does any one know the reason?

### Minimal Reproducible Example

Using Val function:
model = YOLO("/kaggle/input/best-6/best.pt")  # load a custom model
metrics = model.val(data="/kaggle/input/data-obb-2/data_obb_1/data/data/data.yaml", save_json = True, show_labels = True, show_boxes = True) 

Using Predition function on the same test set:
model = YOLO("/kaggle/input/best-6/best.p") 
results = model("/kaggle/input/resize-obb-valid/resized_images", name = "resized_images", show = False, save = True, save_txt= True, show_labels=False, show_conf= False, show_boxes =True)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@nhatdtn this bug report is non-reproducible without the trained models and images, therefore there is no action for us to take at this time.

**Comment by nhatdtn** [comment]
here is link to model, image visualize results using validation code and prediction code
https://drive.google.com/file/d/1ox5dik_JPYMZPQUQO-UYGrvZySajk9tq/view?usp=drive_link

**Comment by glenn-jocher** [comment]
@nhatdtn thank you for providing the link. To better understand the issue, could you please confirm if you are using the latest version of the Ultralytics package? Sometimes discrepancies between validation and prediction results can be due to differences in the package versions or configurations. If the issue persists with the latest version, please let us know, and we'll investigate further.
